syntax = "proto3";

package service;

import "hero.proto";

option java_multiple_files = true;
option java_package = "com.herthrone.service";
option java_outer_classname = "HerthroneProto";
option objc_class_prefix = "HLW";


service Herthrone {
    rpc ListHeroes (ListHeroesRequest) returns (ListHeroesResponse) {}
    rpc ListCards (ListCardsRequest) returns (ListCardsResponse) {}
}

message ListHeroesRequest {
    string name = 1;
}

message ListHeroesResponse {
    repeated Hero heroes = 1;
}

message ListCardsRequest {
    string class_type = 1;
}

message ListCardsResponse {
    repeated string cards = 1;
}
